app/controllers/concerns/flashable.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module Flashable
+  extend ActiveSupport::Concern
+
+  def message_for_notice(invalid_records)
+    msg = "Uploading #{upload_params[:csv_type]} succeeded"
+    msg + (invalid_records.present? ? ' with warnings:' : '.')
+  end
+
+  def errors_for_alert(invalid_records)
+    error_list = invalid_records[0, 15].map do |record|
+      line = record.errors.delete(:base)
```

Comment:
For some files, like Eight-keys there would be 120 "Validation Errors" that aren't really errors. I limit the number of errors reported to the user on the results because 120 would be a lot to display, and because of the nature of the errors being 'free text' embedded in the data rows of the CSV. The error being removed contains the line number of the row causing it and is essentially for display purposes only, and not really part of the model.